src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3841,18 +3976,23 @@ IEnumerator IEnumerable.GetEnumerator()
 
         public IGrouping<TKey, TElement>[] ToArray()
         {
-            IArrayProvider<IGrouping<TKey, TElement>> lookup = Lookup<TKey, TElement>.Create<TSource>(_source, _keySelector, _elementSelector, _comparer);
+            IIListProvider<IGrouping<TKey, TElement>> lookup = Lookup<TKey, TElement>.Create<TSource>(_source, _keySelector, _elementSelector, _comparer);
             return lookup.ToArray();
         }
 
         public List<IGrouping<TKey, TElement>> ToList()
         {
-            IListProvider<IGrouping<TKey, TElement>> lookup = Lookup<TKey, TElement>.Create<TSource>(_source, _keySelector, _elementSelector, _comparer);
+            IIListProvider<IGrouping<TKey, TElement>> lookup = Lookup<TKey, TElement>.Create<TSource>(_source, _keySelector, _elementSelector, _comparer);
             return lookup.ToList();
         }
+
+        public int Count
+        {
+            get { return Lookup<TKey, TElement>.Create<TSource>(_source, _keySelector, _elementSelector, _comparer).Count; }
+        }
```

Comment:
Isn't this going to be really expensive if, for example, it's used in combination with one of the other operators you added that accesses Count as an optimization to size an output array or list, or compute offsets, or things like that?  We'd effectively be building the lookup twice.